Add Airflow hook documentation (#2291)
treeverse/lakeFS: docs/setup/hooks.md

}
```

### Airflow Hooks
Airflow Hook triggers a DAG run in an Airflow installation using [Airflow's REST API](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_dag_run).

> Ends successfully if the DAG was triggered, and fails if it wasn't. 

Reviewer: ```suggestion
The hook run succeeds if the DAG was triggered, and fails otherwise.
```

#### Action file Airflow hook properties

| Property | Description                                     | Data Type | Example                 | Required |
|----------|-------------------------------------------------|-----------|-------------------------|----------|

> | url      | The URL address of the Airflow instance         | String    | "http://localhost:8080" | true     |

Reviewer: ```suggestion
| url      | The URL of the Airflow instance         | String    | "http://localhost:8080" | true     |
```

| dag_id   | The DAG to trigger                              | String    | "example_dag"           | true     |

> | username | The name of the user performing the request     | String    | "admin"                 | true     |

Reviewer: Which user is it? Airflow?

Author: Yes. Done.

| password | The password of the user performing the request | String    | "admin"                 | true     |

> | conf     | DAG run configuration that will be pass as is   | JSON      |                         | false    |

Reviewer: ```suggestion
| conf     | DAG run configuration that will be passed as is   | JSON      |                         | false    |
```

Example:
```yaml
...
hooks:
  - id: trigger_my_dag
    type: airflow
    description: Trigger an example_dag
    properties:
       url: "http://localhost:8000"
       dag_id: "example_dag"
       username: "admin"
       password: "admin"
       conf:
          some: "additional_conf"
...
```

> #### Hook Record in configuration field

Reviewer 2: DAG conf

> lakeFS will add an entry to the configuration property with the event that triggered the action.

Reviewer: Which configuration?

Author: Airflow request, added.

> The key of the record would be `lakeFS_event` and value will match the one described in the previous [schema](#request-body-schema)

Reviewer: ```suggestion
The key of the record will be `lakeFS_event` and the value will match the one described [here](#request-body-schema)
```
